src/main/java/uk/gov/companieshouse/overseasentitiesapi/interceptor/UserAuthenticationInterceptor.java

```diff
@@ -48,18 +52,32 @@ public boolean preHandle(@NonNull HttpServletRequest request, @NonNull HttpServl
             return true;
         }
 
+
         // TokenPermissions should have been set up in the request by TokenPermissionsInterceptor
         final var tokenPermissions = getTokenPermissions(request)
                 .orElseThrow(() -> new IllegalStateException("UserAuthenticationInterceptor - TokenPermissions object not present in request"));
 
+        boolean isRoeUpdateJourneyRequest = isRequestROEUpdate(request);
+
```

Comment:
I think we should introduce a decision point here, based on whether it's the registration or update journey that is being handled by the API end-point. So use the `isRoeUpdateJourneyRequest` value to decide which permission should be checked and only do specific checks for those journeys. Essentially, an `if` ... `else` block.